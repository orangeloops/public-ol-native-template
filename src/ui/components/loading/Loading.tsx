import {observer} from "mobx-react";
import * as React from "react";
import {Text, View} from "react-native";
import RNModal from "react-native-modal";
import {AppStore} from "../../stores/AppStore";
import {BaseComponent} from "../BaseComponent";
import {styles} from "./Loading.styles";

export type LoadingProps = {
  isVisible: boolean;
  animationOutTiming?: number;
};

type LoadingState = {};

@observer
export class Loading extends BaseComponent<LoadingProps, LoadingState> {
  render() {
    const {props} = this;
    const {isVisible} = this.props;

    let animationOutTiming = typeof props.animationOutTiming === "number" ? props.animationOutTiming : AppStore.DEFAULT_OUT_ANIMATION_TIMING;

    return (
      <RNModal isVisible={isVisible} animationOutTiming={animationOutTiming}>
        <View style={styles.container}>
          <Text>{this.formatMessage("Common-loadingText")}</Text>
        </View>
      </RNModal>
    );
  }
}
