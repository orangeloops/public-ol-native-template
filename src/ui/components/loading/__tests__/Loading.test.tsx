import * as React from "react";
import renderer from "react-test-renderer";

import {Loading} from "../Loading";

describe("Loading", () => {
  test("Renders without errors", () => {
    expect(renderer.create(<Loading isVisible={true} />).toJSON()).toMatchSnapshot();
  });
});
