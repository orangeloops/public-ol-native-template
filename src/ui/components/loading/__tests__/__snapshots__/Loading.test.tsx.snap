// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Renders without errors 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "black",
          "borderWidth": 1,
          "height": 80,
          "justifyContent": "center",
          "width": "70%",
        }
      }
    >
      <Text>
        Loading
      </Text>
    </View>
  </View>
</Modal>
`;
